job/server/src/main/java/alluxio/master/meta/JobMasterSync.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.meta;
+
+import alluxio.conf.Configuration;
+import alluxio.grpc.JobMasterMetaCommand;
+import alluxio.grpc.Scope;
+import alluxio.heartbeat.HeartbeatExecutor;
+import alluxio.wire.Address;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicReference;
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * If a master is detected as a standby master. It will set up its MetaMasterSync and manage its own
+ * {@link RetryHandlingMetaMasterMasterClient} which helps communicate with the leader master.
+ *
+ * When running, the standby master will send its heartbeat to the leader master. The leader master
+ * may respond to the heartbeat with a command which will be executed. After which, the task will
+ * wait for the elapsed time since its last heartbeat has reached the heartbeat interval. Then the
+ * cycle will continue.
+ */
+@NotThreadSafe
+public final class JobMasterSync implements HeartbeatExecutor {
```

Comment:
this is totally the same as BlockMasterSync, including how RPCs are designed and arranged